flask/build.gradle

```diff
@@ -0,0 +1,57 @@
+plugins {
+    id 'groovy'
+    id "java-gradle-plugin"
+}
+
+evaluationDependsOn("common")
+evaluationDependsOn("launcher")
+
+dependencies {
+    compileOnly project("common")
+}
+
+FileCollection launcherTar = project("launcher").tasks.named("tar").get().outputs.files
+FileCollection commonJar = project("common").tasks.named("jar").get().outputs.files
```

Comment:
the purpose of the tar is to embed the launcher code inside the Gradle plugin, I cannot simply embed the class files in the plugin jar because otherwise Gradle instrument them with its own annotation when it runs, the goal of the tar archive is basically to keep them out of Gradle reach. I could use a jar file as well but a jar file created with Gradle compresses all the entries so you would end up compressing them twice if you create a jar containing a jar (I know it is a minor issue, but since this archive is both created and extracted by Gradle itself, using a `tar` doesn't require any additional dependency and it only has the advantage of avoiding the double compression)

Reply:
the reason for embedding the launcher code in the gradle plugin is to simplify distribution since it doesn't require us to publish 2 separate artifacts, one for the Gradle plugin and one for the launcher (the launcher code alone would also not be very useful without the Gradle plugin)

Reply:
Caspule _was_ two artifacts of course, except that its Gradle plugin was published by someone else entirely.

I think you are worrying too _far_ much about double-compression. We have lived with it without any problems _for years_ with Corda and Capsule. And the launcher is miniscule in comparison to Corda.